app/ide-desktop/lib/dashboard/src/hooks/debounceCallbackHooks.ts

```diff
@@ -0,0 +1,93 @@
+/**
+ * @file
+ *
+ * This file contains the `useDebouncedCallback` hook which is used to debounce a callback function.
+ */
+import * as React from 'react'
+
+import * as callbackHooks from './eventCallbackHooks'
+import * as unmountEffect from './unmountEffectHooks'
+
+/**
+ * Wrap a callback into debounce function
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function useDebouncedCallback<Fn extends (...args: any[]) => any>(
```

Comment:
nit: `(...args: never) => unknown` should work as a constraint

Reply:
`(...args: never[]) => unknown` works even better :)

Reply:
on a tangent: *technically* `(...args: never) => unknown` is wider - `[]` is a valid value for `never[]`, so when that type is used as a type annotation, `() => {}` would be a valid value of that type
(whereas `: never` means we don't know what it can be called with)

Reply:
Thue,  but `never[]` works better in this situation: https://www.typescriptlang.org/play/?ts=5.5.2#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcA5FDvmQFA0ECuAdvsBE3ExDAOrQDWwJgHMAPADEOWAB4wsTACboAFADp1yKMNQAuTlgBuWKAEo4AXgB8cZvy4B3JpeW5kAGzcAjPPz2SzAN40cCFwuOyo8G5oMADC7m7YBBaYVDCqDKhYSaIBKFq6cAAKmsggWHJQqBJOcAC+ziZ0oXDAycrRkfEeSaq4DFCUTDCBwS1hCd64-GrqnXEJvf2D8uma2k0tdTTb9Mys7HD2AkJiknAycooq6qrrhUyGxgDaALpmVjZMdhCOzq4eKa+OD+OBBFrhJiRODzbqJLDJcypPDpTLZBG5fLaPQlKBlCrGaqSawNZRNMYhNpwDoxOFLAZDEZgimhAFeHyzVSwxYIvoM1Z3AqbULbOpAA